src/LiteDB/Engine/Query/Indexes/IndexInEnumerator.cs

```diff
@@ -0,0 +1,65 @@
+﻿namespace LiteDB.Engine;
+
+internal class IndexInEnumerator : IPipeEnumerator
+{
+    private readonly Collation _collation;
+    private readonly IndexDocument _indexDocument;
+    private readonly IOrderedEnumerable<BsonValue> _values;
+    private readonly int _order;
+
+    private bool _init = false;
+    private bool _eof = false;
+
+    private int _index = -1;
+
+    private IndexEqualsEnumerator _currentIdexer;
+
+    public IndexInEnumerator(
+        BsonArray values,
+        IndexDocument indexDocument,
+        Collation collation,
+        int order)
+    {
+        _values = order == Query.Ascending ? values.OrderBy(x => x) : values.OrderByDescending(x => x);
```

Comment:
A ordem não é dos valores de entrada.... e nesse caso do IN não terá order (pois um IN são multiplos Equals)